From d9b7a2cc54df2d5b5934da5ffa8befded23e842b Mon Sep 17 00:00:00 2001
From: Zhan Lou <zhan.lou@cixtech.com>
Date: Thu, 5 Feb 2026 09:27:23 +0800
Subject: [PATCH] DPTSW-23055: gl: add AFBC formats to glupload caps

Add following AFBC formats to glupload sink caps for end-to-end AFBC
video playback.
  - YUV420_AFBC_8
  - YUV420_AFBC_10
  - YUV422_AFBC_8

Signed-off-by: Zhan Lou <zhan.lou@cixtech.com>
Change-Id: I67f04e4aaa40ec071ed0f5d45259a2c354e2a94a

diff --git a/gst-libs/gst/gl/gstglmemory.h b/gst-libs/gst/gl/gstglmemory.h
index 99c727bee1..7e6f65d344 100644
--- a/gst-libs/gst/gl/gstglmemory.h
+++ b/gst-libs/gst/gl/gstglmemory.h
@@ -68,7 +68,8 @@ GType gst_gl_memory_allocator_get_type(void);
 #define GST_GL_MEMORY_VIDEO_FORMATS_STR \
     "{ RGBA, BGRA, RGBx, BGRx, ARGB, ABGR, xRGB, xBGR, GBRA, GBR, RGBP, BGRP, RGB, BGR, RGB16, BGR16, " \
     "AYUV, VUYA, A444, A422, Y410, I420, YV12, NV12, NV21, NV16, NV61, YUY2, UYVY, Y210, Y41B, " \
-    "Y42B, Y444, GRAY8, GRAY16_LE, GRAY16_BE, ARGB64, A420, AV12, NV12_16L32S, NV12_4L4, RBGA, v210" \
+    "Y42B, Y444, GRAY8, GRAY16_LE, GRAY16_BE, ARGB64, A420, AV12, NV12_16L32S, NV12_4L4, RBGA, v210, " \
+    "YUV420_AFBC_8, YUV420_AFBC_10, YUV422_AFBC_8 " \
     GST_GL_MEMORY_VIDEO_EXT_FORMATS "}"
 
 /**
